File: backend/src/main/java/com/rodrigocoelhoo/lifemanager/food/dto/RecipeDTO.java
package com.rodrigocoelhoo.lifemanager.food.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public record RecipeDTO(

    @NotBlank(message = "Recipe name cannot be blank")
    @Size(max = 50, message = "Name cannot exceed 50 characters")
    String name,

    @Valid
    @NotNull(message = "Ingredient list cannot be null, but can be empty")
    List<RecipeIngredientDTO> ingredients
) { }

File: frontend/src/services/nutrition/recipe/recipe.dto.ts
import type { IngredientResponseDTO } from "../ingredient/ingredient.dto";

export interface RecipeIngredientResponseDTO {
	ingredient: IngredientResponseDTO;
	amount: number;
	unit: string;
}

export interface RecipeDetailsDTO {
	id: number;
	name: string;
	ingredients: RecipeIngredientResponseDTO[];	
}

export interface RecipeIngredientDTO {
	id: number;
	amount: number;
	unit: string;
}

export interface RecipeDTO {
	name: string;
	ingredients: RecipeIngredientDTO[];
}

export const Unit = {
	MG: "mg",
	G: "g",
	KG: "kg",
	ML: "ml",
	L: "l",
} as const;
export type Unit = keyof typeof Unit;